README.md

```diff
@@ -85,6 +94,38 @@ nginx::resource::mailhost { 'domain1.example':
 }
 ```
 
+### Convert upstream members from Array to Hash
+
+The datatype Array for members of a nginx::resource::upstream is replaced by a Hash. The following configuration is no longer valid:
+
+```puppet
+nginx::resource::upstream { 'puppet_rack_app':
+  members => [
+    'localhost:3000',
+    'localhost:3001',
+    'localhost:3002',
+  ],
+}
+```
+
+From now on, the configuration must look like this:
+
+```puppet
+nginx::resource::upstream { 'puppet_rack_app':
+  members => {
+    'localhost:3000':
+      server => 'localhost'
+      port   => 3000
```

Comment:
At first glance this can be a little confusing. My first question would be "Why not parse the `localhost:3000` into server and port?"

Could we potentially make it an array of hashes so that it looks more like:
``` puppet
nginx::resource::upstream { 'puppet_rack_app':
  members => [
    {
      server => 'localhost',
      port    => 3000,
    },
    {
      server => 'localhost',
      port    => 3001,
    },
    {
      server => 'localhost',
      port    => 3002,
    },
  ]
```
Does that work or does the parameter rely on each member to have a unique name?

Reply:
Mainly cause Unix sockets are accepted too. We would first have to see if it is a valid Unix socket and only if not split at the colon. After that we need to check the parts if they are valid. We almost have to implement the used datatype in code. And at the end we have the problem to name the resources for the defined type which is necessary for exporting upstream members. That's why we need a unique name here.
Should I change the README.md file and use keys that do not consist of server and port? Maybe it's a good idea to use the keys as default for the comment.